ledger/complete/mtrie/trie/trie.go

```diff
@@ -379,70 +385,82 @@ type updateResult struct {
 //     (excluding the bit at index headHeight)
 //   - paths are NOT duplicated
 func update(
-	nodeHeight int, parentNode *node.Node,
-	paths []ledger.Path, payloads []ledger.Payload, compactLeaf *node.Node,
-	prune bool,
+	nodeHeight int, // the height of the node during traversing the subtree
+	currentNode *node.Node, // the current node on the travesing path, if it's nil it means the trie has no node on this path
```

Comment:
Rename it from `parentNode` to `currentNode` to avoid confusion.

I think originally we wanted to use "parentNode" to mean this node is the parent of its two child nodes. However, in a binary tree structure, a node has a single parent and two child nodes, naming the node itself as "parent node" creates some confusion.